src/MessageTranslator.cpp

```diff
@@ -0,0 +1,281 @@
+
+    return umsg;
+}
+
+/**
+  * @brief @return @see @ref MessageTranslator::translateSomeipToUMsgForResponse()
+  * @param[in] someIpMessage @see @ref MessageTranslator::translateSomeipToUMsgForResponse().
+  * @param[in] uri @see @ref MessageTranslator::translateSomeipToUMsgForResponse().
+  * @param[in] uuid @see @ref MessageTranslator::translateSomeipToUMsgForResponse().
+  */
+std::shared_ptr<UMessage> MessageTranslator::translateSomeipToUMsgForResponse(
+    std::shared_ptr<message> const &someIpMessage,
+    std::shared_ptr<uprotocol::utransport::UMessage> originalRequestMsg) const {
+
+    std::shared_ptr<UMessage> uMsg = std::make_shared<UMessage>();
+
+    //UUID const &generatedUUID = uprotocol::uuid::Uuidv8Factory::create(); //TODO : Remove this comment once zenoh uTransport fix the issue.
+
+    // Build Payload
+    std::shared_ptr<UPayload> uPayload = buildUPayload(someIpMessage);
+
+    // Build Attributes
+    //TODO: attributes.id field of response should be newly generated but currently it is not as per uProtocol 1_5_7
+    //so need to set original resquet uuid to attributes.id as a workaround. This needs to be fixed in zenoh uTransport
+    UMessageType const type = convertSomeipToUMessageType(message_type_e::MT_RESPONSE);
+    UAttributesBuilder builder(
+        originalRequestMsg->attributes().source() ,
+        originalRequestMsg->attributes().id(),
+        type, originalRequestMsg->attributes().priority());
+
+    builder.setSink(originalRequestMsg->attributes().sink());
+    builder.setReqid(originalRequestMsg->attributes().id());
+    std::shared_ptr<UAttributes> attributes = std::make_shared<UAttributes>(builder.build());
+
+
+    uMsg->setPayload(*uPayload);
+    uMsg->setAttributes(*attributes);
+
+    return uMsg;
+}
+
+/**
+ * @brief @return @see @ref MessageTranslator::translateSomeipToUMsgForRequest()
+ * @param[in] someIpMessage @see @ref MessageTranslator::translateSomeipToUMsgForRequest().
+ * @param[in] uEntity @see @ref MessageTranslator::translateSomeipToUMsgForRequest().
+ * @param[in] uResource @see @ref MessageTranslator::translateSomeipToUMsgForRequest().
+ */
+std::shared_ptr<UMessage> MessageTranslator::translateSomeipToUMsgForRequest(
+    std::shared_ptr<message> const &someIpMessage,
+    UEntity const &uEntity,
+    UResource const &uResource) const {
+
+    LogInfo("{} serviceid[0x{:x}] methodId[0x{:x}]",
+            __FUNCTION__, someIpMessage->get_service(), someIpMessage->get_method());
+
+    UUID const &generatedUUID = uprotocol::uuid::Uuidv8Factory::create();
+
+    std::shared_ptr<UPayload> uPayload = buildUPayload(someIpMessage);
+
+    UAuthority uauthority;
+    std::shared_ptr<UUri> uriPtr = std::make_shared<UUri>();
+    uriPtr->mutable_authority()->CopyFrom(uauthority);
+    uriPtr->mutable_entity()->CopyFrom(uEntity);
+    uriPtr->mutable_resource()->CopyFrom(uResource);
+
+    //TODO: Currenty assigning source URI same value as sink URI, since it is not been used anywhere
```

Comment:
In the future, in lieu of these workarounds, we need to fix them in the client library as this has been an open bug for a while.